src/components/Toastify.tsx

```diff
@@ -0,0 +1,114 @@
+/* eslint-disable @typescript-eslint/restrict-template-expressions */
+import { Box, Link } from "@mui/material"
+import React, { ReactText } from "react"
+
+import LaunchIcon from "@mui/icons-material/Launch"
+import { getEtherscanLink } from "../utils/getEtherscanLink"
+import { toast as toastify } from "react-toastify"
+
+type ToastStatus = "success" | "info" | "error"
+type TxType =
+  | "tokenApproval"
+  | "deposit"
+  | "swap"
+  | "withdraw"
+  | "claim"
+  | "migrate"
+  | "stake"
+  | "unstake"
+
+export const enqueuePromiseToast = (
+  promy: Promise<unknown>,
+  type: TxType,
+  additionalData?: { tokenName?: string; poolName?: string },
+): Promise<unknown> => {
+  const renderPendingContentBasedOnType = (type: string) => {
+    switch (type) {
+      case "deposit":
+        return "Deposit Initiated"
+      case "tokenApproval":
+        return `Approve ${additionalData?.tokenName} spend`
+      case "swap":
+        return "Swap Initiated"
+      case "withdraw":
+        return "Withdraw Initiated"
+      case "claim":
+        return "Claim Initiated"
+      case "migrate":
+        return "Migrate Initiated"
+      case "stake":
+        return "Stake Initiated"
+      case "unstake":
```

Comment:
what do you think of making these constants eg `TXN_TYPES.STAKE` to prevent me from typing `wtihdraw` and showing the wrong UI?

Reply:
since these are typed, tsc will prevent misspelling of potential arguments you can pass. it will also auto complete as soon as you start typing out the fn i.e:

<img width="658" alt="image" src="https://user-images.githubusercontent.com/5507707/156643900-a6099b6b-2372-4706-a787-a53839a81e09.png">
<img width="1251" alt="image" src="https://user-images.githubusercontent.com/5507707/156643971-140fcb58-8418-45ec-b11e-d2531ae023a8.png">